### DISCUSSION TITLE
console log prints value of assigned object instead of original and shows later value

### DISCUSSION THREAD

**Discussion by VitalJeevanjot** [quasarframework/quasar#12767] [CLI - Electron mode] [answered]
hello,

```js
const tenc = new TextEncoder()
const tdec = new TextDecoder()

const some_val = 'hello'

const hello = {
  a: 1,
  b: tenc.encode(some_val),
}

console.log(hello)

let k = null
k = hello
k.b = tdec.decode(k.b)

console.log(k)
```

If you run the above code from the quasar current version dev commands, You will get the value from `console.log(hello)` of `k` (more specifically, It shows value as a string, Instead of an Array when you expand the console log result but when not expended then it shows the correct datatype. I tried from both modes:- electron & browsers (chrome & firefox). 

```
Running @quasar/cli v1.2.2
Running @quasar/app v3.3.3
```

**Comment by MilosPaunovic** [comment, marked as answer]
This does not seem Quasar related.

**Reply by VitalJeevanjot** [reply]
I actually tried to create this behavior online through multiple compilers but seems like it can only be created through webpack builders like quasar as I am using (maybe others but didn't try). So any explanation will be great.

**Reply by MilosPaunovic** [reply]
You've said it yourself - if it can be created only using webpack, then it's repo is the place to post this question. But logging stuff to the console might be browser specific and related to how browser console itself works.

**Reply by VitalJeevanjot** [reply]
ah I see,
Just tried on the browser console

**Check the outputs**

![image](https://user-images.githubusercontent.com/40867747/158014467-21ad31e7-f3bf-48a7-81b9-52eae2eac1fc.png)

It does seem like a browser console thing...

Online compilers

![image](https://user-images.githubusercontent.com/40867747/158014731-44fc389e-a6ed-45ce-b402-350fb530493c.png)

Confused right now as to why different results because if it has to work as pass by reference then it should do it everywhere...

**Reply by VitalJeevanjot** [reply]
You are right though not Quasar specific issue as you said.
Thank you.

**Reply by VitalJeevanjot** [reply]
Anybody here later to check (if this stays)
Then please refer to this for your answer https://stackoverflow.com/questions/38660832/console-log-of-element-children-shows-0-length-but-has-three-entries-when-expand
